designs/2021-plugin-rule-replacements/README.md

```diff
@@ -0,0 +1,75 @@
+- Repo: eslint/eslint, eslint/eslintrc
+- Start Date: 2021-01-23
+- RFC PR: (to be filled in)
+- Authors: mykhalov
+
+# Allow plugins to define rule replacements
+
+## Summary
+
+Provide better error message for removed rules by allowing plugins to define rule replacements.
+
+## Motivation
+
+ESLint provides great developer experience around removed rules:
+
+```
+  1:1  error  Rule 'generator-star' was removed and replaced by: generator-star-spacing  generator-star
+```
+
+Until now this hasn't been the case with plugins:
+
+```
+  1:1  error  Definition for rule '@typescript-eslint/camelcase' was not found  @typescript-eslint/camelcase
+```
+
+Although `@typescript-eslint/camelcase` [has been removed in favour of another rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/camelcase.md), there is no way for the plugin to explain this.
+
+## Detailed Design
+
+When reading plugin definition, look for `"replacements"` property next to `"rules"`:
```

Comment:
This section should explain how you plan to implement this. Please look at the ESLint code and figure out which files would need to change. There are plenty of examples of already-merged RFCs in this repo to see what is expected.